server/utils/tans.py

```diff
@@ -83,9 +91,23 @@ def uniques(n: int, length: int = 5) -> list[Tan]:
     """
     if n > possible_tans(length):
         raise ValueError(
-            f"Cannot generate {n} unique TANs of length {length}. Maximum possible TANs is {possible_tans(length)}."
+            f"Cannot generate {n} unique TANs of length {length}. Maximum possible TANs of this length are "
+            f"{possible_tans(length)}."
         )
+    # Retrieve existing tans
     tans = set()
-    while len(tans) < n:
-        tans.add(Tan(length))
-    return list(tans)
+    with create_app(csrf, db).app_context():
+        db_tans = [tan for tan, in db.session.query(models.Player.tan).all()]
```

Comment:
.all() call not necessary. .query already gives back an iterable: https://stackoverflow.com/questions/1078383/sqlalchemy-difference-between-query-and-query-all-in-for-loops